src/vmm/src/vstate/memory.rs

```diff
@@ -79,6 +79,13 @@ where
         track_dirty_pages: bool,
     ) -> Result<Self, MemoryError>;
 
+    /// Creates a GuestMemoryMmap from raw regions with guard pages.
+    fn from_raw_regions_file(
```

Comment:
can make the 1st line of the commit message more descriptive by referring to adding this method to the trait.

Reply:
Added